cheroot/server.py

```diff
@@ -1805,7 +1793,22 @@ def bind(self, family, type, proto=0):
                 # this machine's TCP stack
                 pass
 
+        # So we can reuse the socket...
```

Comment:
I dislike how this functionality is now embedded in the bind method and straddling the bind call.

I'm inclined to leave the unlink operation in prepare socket.

Reply:
Please take a look at this comment of the reporter: https://github.com/cherrypy/cheroot/issues/93#issuecomment-417620472
It looks like these things are quite tightly coupled. Also, it's needed exactly around `socket.bind()`.

What I don't like even more is that bypassing of the bind_addr is done via implicit assignment to the self object attribute.